src/main/java/seedu/duke/Storage.java

```diff
@@ -108,19 +114,44 @@ public static void loadFromFile(InternshipList internshipList) {
                 }
 
                 // Parse favourite internships
-                String[] parts = line.substring("FAVOURITES:".length()).trim().split(" ");
-                for (String id : parts) {
-                    int favInternshipId = Integer.parseInt(id);
-                    int favInternshipIndex = favInternshipId - 1;
-                    Internship favInternship = internshipList.internships.get(favInternshipIndex);
-                    internshipList.favouriteInternships.add(favInternship);
+                if (!internshipList.favouriteInternships.isEmpty()) {
+                    String[] parts = line.substring("FAVOURITES:".length()).trim().split(" ");
+                    for (String id : parts) {
+                        int favInternshipId = Integer.parseInt(id);
+                        int favInternshipIndex = favInternshipId - 1;
+                        Internship favInternship = internshipList.internships.get(favInternshipIndex);
+                        internshipList.favouriteInternships.add(favInternship);
+                    }
                 }
+
             }
         } catch (IOException e) {
             System.out.println("Error while loading tasks: " + e.getMessage());
         }
     }
 
+    private static boolean isValidFormat(String line) {
+        if (line.startsWith("FAVOURITES:")) {
+            return line.matches("FAVOURITES:( \\d+)*");
+        }
+        return line.matches(
+                "\\d+ \\| [^|]+ \\| [^|]"
+                        + "+ \\| \\d{2}/\\d{2}/\\d{4} \\| \\d{2}/\\d{2}/\\d{4} \\| [^|]"
+                        + "+ \\| [^|]+ \\| .*");
+    }
+
+    private static void resetFile() {
+        File file = new File(FILE_PATH);
```

Comment:
Maybe we can just ignore all the data stored, then just override it instead? Rather than delete and recreate the file